aten/src/ATen/native/vulkan/api/Context.cpp

```diff
@@ -78,25 +78,42 @@ VkQueue acquire_queue(
 
 } // namespace
 
-void Context::Deleter::operator()(const VkDevice device) const {
-  // No VK_CHECK.  Don't want an exception thrown in the destructor.
-  vkDeviceWaitIdle(device);
-  vkDestroyDevice(device, nullptr);
-}
-
 Context::Context(const Adapter& adapter)
     : adapter_(adapter),
       device_(
           create_device(
               adapter.handle,
               adapter.compute_queue_family_index),
-          Deleter{}),
+          &VK_DELETER(Device)),
       queue_(acquire_queue(device(), adapter.compute_queue_family_index)),
       command_(gpu()),
       shader_(gpu()),
       pipeline_(gpu()),
       descriptor_(gpu()),
       resource_(gpu()) {
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+      device_,
+      "Invalid Vulkan device!");
+}
+
+Context::~Context() {
+  try {
+    flush();
+  }
+  catch (...) {
```

Comment:
Is it by design that it will catch everything silently even for debug?

Thinking that at least for debug we might want to see the errors or will they show up at some other listener?